### DISCUSSION TITLE
Need to know how can we have a fixed height and width for openseadragon container

### DISCUSSION THREAD

**Discussion by suganya27302** [openseadragon/openseadragon#2484] [Q&A]
<div ref={containerref} style="width: 100%; height: 100%;">
div id="openseadragon1" style="width: 800px; height: 600px;"></div>
</div>
<script src="/openseadragon/openseadragon.min.js"></script>
<script type="text/javascript">
    var viewer = OpenSeadragon({
        id: "openseadragon1",
        prefixUrl: "/openseadragon/images/",
        tileSources: "/path/to/my/image.dzi"
    });
</script>

If I provide a width like this it is not scable across the screens. So I tried to provide the width with JS style 
container = containerref.current!
width: container.offsetWidth,
height: container.offsetHeight,
But its width does not take the entire container on exiting from the fullscreen. The image shrinks. Can someone help with this problem?

**Comment by iangilman** [comment]
OpenSeadragon always grows to fill the container you give it, so whatever styling you want to do on the container element should work. You can test this even without OSD... Give your element a background color and style it to fill the space you want.

**Reply by suganya27302** [reply]
```JSX
parentContainer{
width:100%;
heigth:100%;
}
const getOpenSeaDragonContainerStyle = () => {
const seaDragonContainer = parentContainerRef.current!;
const containerStyle: React.CSSProperties = {
width: seaDragonContainer.offsetWidth,
height: seaDragonContainer.offsetHeight,
};
return containerStyle;
};
<div ref={parentContainerRef} className={styles.parentContainer}>
<div style={getOpenSeaDragonContainerStyle()}>
   <OpenSeadragonAnalyzer
   imageSrc={src}
   showToolBar={false}
   openSeadragonOptions={{
   homeFillsViewer: true,
   }}
   />
</div>
```

Instead of providing a fixed height and width directly. Based on the parent div reference calculate the width and height, then provide it to the openSeaDragon div. 
On the initial render of the page, when we enter into fullscreen and exit from the fullscreen, the offsetWidth is 350-360px. So the image shrinks.
How the width and height for the image is calculated within the openseaDragon?

**Reply by iangilman** [reply]
Are you talking about the viewer (the entire space OSD uses to render in) or just the image (which can zoom and pan around within the viewer)? Are you saying that the image gets zoomed out after returning from fullscreen? Something like that might happen because the aspect ratio of fullscreen is different than the aspect ratio in the non-fullscreen viewer. What version of OSD are you using?

**Reply by suganya27302** [reply]
I am mentioning about the Viewer (the entire space OSD uses to render the image)

Actual image in the viewer:
![image](https://github.com/openseadragon/openseadragon/assets/77053019/ec8557d9-ab4b-4558-925d-683da7861436)

After exiting from the full screen, the image looks like this
![image](https://github.com/openseadragon/openseadragon/assets/77053019/079368ea-0f48-49ba-9074-c37b8f9e55b2)

The OSD version we are using is  4.0.0

**Reply by iangilman** [reply]
Okay, thank you. I'm not sure why this is happening. My best guess is it has something to do with the fullscreen behavior discussed in https://github.com/openseadragon/openseadragon/discussions/2486. Did you get that sorted out?

If you'd like us to take a look at your issue, can you share a link to the page, or create a simplified repro?

**Reply by suganya27302** [reply]
I couldn't able to share the link due to confidentiality. I kind of did a quick fix, not an exact fix

**Reply by iangilman** [reply]
Fair enough. Hopefully it's good enough? If not, you could still make a cut-down repro with dummy images.
